lib/elixir_auth_github.ex

```diff
@@ -82,13 +82,22 @@ defmodule ElixirAuthGithub do
     |> check_authenticated
   end
 
-  defp check_authenticated(%{"access_token" => access_token}) do
+  defp check_authenticated(%{"access_token" => access_token, "scope" => scope}) do
     access_token
     |> get_user_details
+    |> set_scope(scope)
   end
 
   defp check_authenticated(error), do: {:error, error}
 
+  defp set_scope({:ok, user_details}, scope) do
```

Comment:
I would actually make this a `def` instead of `defp` so that it could be tested independently
to make sure that it degrades gracefully. e.g. if `scope` is `nil` does it still work?

Looks like you're handling the case of `_scope` below but it's not _immediately_ obvious. 💭